### ISSUE TITLE
Questions about TSDAE

### ISSUE THREAD

**Issue by MatthewCYM** [UKPLab/sentence-transformers#1305]
Hi @nreimers ,

Thanks for publishing and sharing the TSDAE approach.

I am a little bit confused about the experiment on domain adaptation.

May I ask what's the training objectives of TSDAE on NLI+STS for the domain adaptation? Is it still unsupervised or supervised?

Yiming

**Comment by nreimers** [comment]
Hi,
@kwang2049 can answer this better than me :)

**Comment by kwang2049** [comment]
Hi Yiming @MatthewCYM, thanks for your attention and question!

If I understood you correctly, you might want to figure out the training procedure of the models under the group "Domain adaptation: target task → NLI+STS" in the main result table (Table 2).

Taking AskUbuntu as an example, we did the following 4 steps in order:
(1) Initialized with bert-base-uncased;
(2) Unsupervised training on AskUbuntu with the **TSDAE objective** (unsupervised);
(3) Supervised training on the NLI data with **cross-entropy loss** (yes it is supervised);
(4) Supervised training on the STSb data with **MSE loss** (yes it is also supervised).

So basically we wanted to use the **unlabeled** data from the **target** task/domain and the **labeled** data from the **source** domain, and the **goal** is to optimize the performance on the **target** task.

(And the group "Domain adaptation: NLI+STS → target task" simply reverses the training order, which we showed performs rather poorly.)

Hope this can answer your question and welcome to any further discussion&questions!

**Comment by MatthewCYM** [comment]
Hi @kwang2049,

Thank you for the reply,

May I ask why you choose cross-entropy loss for the NLI data, instead of using current SOTA supervised sentence embedding training objectives, e.g., sbert, simcse?

Yiming

**Comment by nreimers** [comment]
Hi @MatthewCYM 
We used cross-entropy loss with in-batch negatives, which is another name for MultipleNegativesRankingLoss: https://www.sbert.net/docs/package_reference/losses.html#multiplenegativesrankingloss

Sometimes it is also known as InfoNCE loss.

It is a rather old loss for learning embeddings (https://arxiv.org/abs/1705.00652) and has been used to learn many embedding models including Universal Sentence Encoder (USE) and the second version of the SBERT models (like paraphrase-*, all-*, msmarco-* models etc). 

The SimCSE authors re-invented the same loss, gave it a new name, and claimed to be new. But it is the most common loss function to train embedding models since 2017.

**Comment by MatthewCYM** [comment]
Hi @nreimers 

Thank you for the explanation. Previously, I thought the supervised training objective is standard fine-tuning for sequence classification. Now I understand the training process better.

Regards,
Yiming
